apps/odk_publish/forms.py

```diff
@@ -302,8 +319,90 @@ class Meta:
             "value": TextInput,
         }
 
+    def __init__(self, *args, **kwargs):
+        valid_template_variables = kwargs.pop("valid_template_variables", None)
+        super().__init__(*args, **kwargs)
+        if valid_template_variables is not None:
+            self.fields["template_variable"].queryset = valid_template_variables
+
 
 ProjectTemplateVariableFormSet = forms.models.inlineformset_factory(
     Project, ProjectTemplateVariable, form=ProjectTemplateVariableForm, extra=0
 )
 ProjectTemplateVariableFormSet.deletion_widget = CheckboxInput
+
+
+class OrganizationForm(PlatformFormMixin, forms.ModelForm):
+    class Meta:
+        model = Organization
+        fields = ["name", "slug"]
+        widgets = {
+            "name": TextInput,
+            "slug": TextInput,
+        }
+
+
+class CleanOrganizationInvitationMixin:
+    """Similar to django-invitation's CleanEmailMixin, but checks for invitations
+    to the same organization.
+    """
+
+    def validate_invitation(self, email, organization):
+        if OrganizationInvitation.objects.all_valid().filter(
+            email__iexact=email, organization=organization, accepted=False
+        ):
+            raise AlreadyInvited
+        elif OrganizationInvitation.objects.filter(
+            email__iexact=email, organization=organization, accepted=True
+        ):
+            raise AlreadyAccepted
+        else:
+            return True
+
+    def clean(self):
+        email = self.cleaned_data["email"]
+        email = get_invitations_adapter().clean_email(email)
+        if hasattr(self, "organization"):
+            organization = self.organization
+        else:
+            organization = self.cleaned_data["organization"]
+
+        errors = {
+            "already_invited": f"This e-mail address has already been invited to {organization}.",
+            "already_accepted": f"This e-mail address has already accepted an invite to {organization}.",
```

Comment:
I ran into an edge case I believe. If I accept an invite, leave the org, and try to invite the same email, the Send an invite form shows a validation error: "this e-mail address has already accepted an invite".

Similarly, I can invite a user that's already in an organization.

Maybe an enhancement ticket for these? What do you think?